boards/xtensa/intel_adsp_cavs25/doc/index.rst

```diff
@@ -0,0 +1,409 @@
+    kernels.
++ ``CONFIG_MODVERSIONS=n`` - Allow modules to be built and installed
+    from modified "dirty" build trees.
+
+Now build your kernel just as you would any other:
+
+.. code-block:: console
+
+    dev$ make olddefconfig     # Or otherwise update .config
+    dev$ make bzImage modules  # Probably want -j<whatever> for parallel build
+
+The modules you can copy directly to the (now writable) rootfs on the
+device.  Note that this filesystem has very limited space (it's
+intended to be read only), so the INSTALL_MOD_STRIP=1 is absolutely
+required, and you may find you need to regularly prune modules from
+older kernels to make space:
+
+.. code-block:: console
+
+    dev$ make INSTALL_MOD_PATH=mods INSTALL_MOD_STRIP=1 modules_install
+    dev$ (cd mods/lib/modules; tar cf - .) | ssh root@cros '(cd /lib/modules; tar xfv -)'
+
+Pack and Install ChromeOS Kernel Image
+======================================
+
+The kernel bzImage file itself needs to be signed and packaged into a
+ChromeOS vboot package and written directly to the kernel partition.
+Thankfully the tools to do this are shipped in Debian/Ubuntu
+repositories already:
+
+.. code-block:: console
+
+    $ sudo apt install vboot-utils vboot-kernel-utils
+
+Find the current kernel partition on the device.  You can get this by
+comparing the "kernel_guid" command line parameter (passed by the
+bootloader) with the partition table of the boot drive, for example:
+
+.. code-block:: console
+
+    dev$ KPART=`ssh root@cros 'fdisk -l -o UUID,Device /dev/nvme0n1 | \
+                               grep -i $(sed "s/.*kern_guid=//" /proc/cmdline \
+                                         | sed "s/ .*//") \
+                               | sed "s/.* //"'`
+    dev$ echo $KPART
+    /dev/nvme0n1p4
+
+Extract the command line from that image into a local file:
+
+.. code-block:: console
+
+    dev$ ssh root@cros vbutil_kernel --verify /dev/$KPART | tail -1 > cmdline.txt
+
+Now you can pack a new kernel image using the vboot tooling.  Most of
+these arguments are boilerplate and always the same.  The keys are
+there because the boot requires a valid signature, even though as
+configured it won't use it.  Note the cannot-actually-be-empty dummy
+file passed as a "bootloader", which is a holdover from previous ROM
+variants which needed an EFI stub.
+
+.. code-block:: console
+
+    dev$ echo dummy > dummy.efi
+    dev$ vbutil_kernel --pack kernel.img --config cmdline.txt \
+           --vmlinuz arch/x86_64/boot/bzImage \
+           --keyblock /usr/share/vboot/devkeys/kernel.keyblock \
+           --signprivate /usr/share/vboot/devkeys/kernel_data_key.vbprivk \
+           --version 1 --bootloader dummy.efi --arch x86_64
+
+You can verify this image if you like with "vbutil_kernel --verify".
+
+Now just copy up the file and write it to the partition on the device:
+
+.. code-block:: console
+
+    $ scp kernel.img root@cros:/tmp
+    $ ssh root@cros dd if=/tmp/kernel.img of=/dev/nvme0n1p4
```

Comment:
Updating modules?

Reply:
Happened a few steps above right after make MODULES_INSTALL.  ChromeOS has no initrd, you just drop the modules in the root filesystem.  Maybe it would be clearer to split the step and do that here, though.